zio-schema/shared/src/main/scala/zio/schema/Schema.scala

```diff
@@ -488,6 +491,38 @@ object Schema extends TupleSchemas with RecordSchemas with EnumSchemas with Sche
       b.makeTraversal(self, as)
   }
 
+  final case class Dynamic(override val annotations: Chunk[Any] = Chunk.empty) extends Schema[DynamicValue] {
+    override type Accessors[Lens[_, _], Prism[_, _], Traversal[_, _]] = Unit
+
+    /**
+     * The default value for a `Schema` of type `A`.
+     */
+    override def defaultValue: Either[String, DynamicValue] =
+      Right(DynamicValue.NoneValue)
+
+    /**
+     * Returns a new schema that with `annotation`
+     */
+    override def annotate(annotation: Any): Schema[DynamicValue] =
+      this.copy(annotations = annotations :+ annotation)
+
+    override def makeAccessors(b: AccessorBuilder): Unit = ()
+  }
```

Comment:
Not sure I follow what the purpose of this is. `DynamicValue` is already an ADT which has an implicit schema associated with it.

Reply:
I think what this gives us is to handle dynamic values specially in codecs. But I did not take any advantage of this in this PR.

So in theory for example in JSON you could probably encode a dynamic value as a JSON subtree without encoding it with the more verbose dynamic value ADT. 
But maybe that would loose some important type information and not possible anyway so in this case this really does not make much sense.

I'm ok with removing it, but I think conceptually it is nice to have it as a special schema, as described in the linked issue (https://github.com/zio/zio-schema/issues/192)

Reply:
@jdegoes 

It makes sense from the standpoint of encoders, I'm just concerned that it ends up creating weird edge cases and unexpected error conditions because `Dynamic` doesn't really describe anything so there is no way to handle, for example, diffing. And without a `Schema` `DynamicValue` doesn't seem like a particularly useful construct. If we want to handle arbitrary JSON, could we just have a `Schema` for the `zio.json.ast.Json`? And then maybe a constructor for `DynamicValue` that takes a JSON AST?